pkg/networkmanager/firewall.jsx

```diff
@@ -181,67 +333,95 @@ class AddServicesBody extends React.Component {
         // hide already enabled services
         if (services)
             services = services.filter(s => !firewall.enabledServices.has(s.id));
-
-        let body;
-        if (this.state.services) {
-            body = (
-                <Modal.Body id="add-services-dialog">
-                    <table className="form-table-ct">
-                        <tbody>
-                            <tr>
-                                <td>
-                                    <label htmlFor="filter-services-input" className="control-label">
-                                        {_("Filter Services")}
-                                    </label>
-                                </td>
-                                <td>
-                                    <SearchInput id="filter-services-input"
-                                                 className="form-control"
-                                                 onChange={this.onFilterChanged} />
-                                </td>
-                            </tr>
-                        </tbody>
-                    </table>
-                    <ul className="list-group dialog-list-ct">
-                        {
-                            services.map(s => (
-                                <li key={s.id} className="list-group-item">
-                                    <label>
-                                        <input data-id={s.id}
-                                               type="checkbox"
-                                               checked={this.state.selected.has(s.id)}
-                                               onChange={this.onToggleService} />
-                                        &nbsp;
-                                        <span>{s.name}</span>
-                                    </label>
-                                </li>
-                            ))
-                        }
-                    </ul>
-                </Modal.Body>
-            );
-        } else {
-            body = (
-                <Modal.Body id="add-services-dialog">
-                    <div className="spinner spinner-lg" />
-                </Modal.Body>
-            );
-        }
-
+        var addText = this.state.custom ? _("Add Ports") : _("Add Services");
         return (
             <Modal id="add-services-dialog" show onHide={this.props.close}>
                 <Modal.Header>
-                    <Modal.Title> {`Add Services`} </Modal.Title>
+                    <Modal.Title> {addText} </Modal.Title>
                 </Modal.Header>
                 <div id="cockpit_modal_dialog">
-                    {body}
+                    <Modal.Body id="add-services-dialog">
```

Comment:
Can you please move Garrett's commits to the front, and squash this and the last commit? This makes it easier to review, and provides a more logical view on master.